File: College/Assignments/ACPL/Python/8.py
import os

def word_count_file(input_file, output_file):
    with open(input_file, 'r') as f:
        text = f.read().split()
    count = {}
    for word in text:
        count[word] = count.get(word, 0) + 1
    with open(output_file, 'w') as f:
        for word in sorted(count):
            f.write(f"{word}: {count[word]}\n")

if __name__ == "__main__":
    script_directory = os.path.dirname(__file__)
    input_file = os.path.join(script_directory, "8_Input_file.txt")
    output_file = os.path.join(script_directory, "8_Output_file.txt")
    word_count_file(input_file, output_file)

File: College/Assignments/ACPL/CPP/8.cpp
#include <fstream>
#include <map>
#include <sstream>
using namespace std;

class Solution{
public:
    void word_count_file(string in_file, string out_file) {
        ifstream in(in_file);
        ofstream out(out_file);
        map<string, int> count;
        string word;
        while (in >> word) count[word]++;
        for (auto& p : count)
            out << p.first << ": " << p.second << endl;
    }
};

int main(){
    Solution sol;
    string in_file = R"(./8_Input_file.txt)";
    string out_file = R"(./8_Output_file.txt)";
    sol.word_count_file(in_file, out_file);
    return 0;
}